src/screens/ChangeWalletSheet.js

```diff
@@ -0,0 +1,266 @@
+      const wallet = wallets[wallet_id];
+      const account = wallet.addresses.find(
+        account => account.address === address
+      );
+
+      navigate(Routes.EXPANDED_ASSET_SCREEN, {
+        address,
+        asset: [],
+        onCloseModal: async args => {
+          if (args) {
+            const newWallets = { ...wallets };
+            if (args.name) {
+              newWallets[wallet_id].addresses.some((account, index) => {
+                if (account.address === address) {
+                  newWallets[wallet_id].addresses[index].label = args.name;
+                  newWallets[wallet_id].addresses[index].color = args.color;
+                  if (selectedWallet.id === wallet_id) {
+                    dispatch(walletsSetSelected(newWallets[wallet_id]));
+                  }
+                  return true;
+                }
+                return false;
+              });
+              await dispatch(walletsUpdate(newWallets));
+            }
+          }
+        },
+        profile: {
+          color: account.color,
+          name: account.label || ``,
+        },
+        type: 'wallet_profile_creator',
+      });
+    },
+    [dispatch, navigate, selectedWallet.id, wallets]
+  );
+
+  const onEditWallet = useCallback(
+    (wallet_id, address, label) => {
+      const wallet = wallets[wallet_id];
+      if (!selectedWallet) return;
+
+      let isDeletable = false;
+      if (Object.keys(wallet).length > 1 && selectedWallet.id !== wallet_id) {
+        isDeletable = true;
+      }
+
+      const buttons = ['Rename Wallet'];
+      if (isDeletable) {
+        buttons.push('Delete Wallet');
+      }
+      buttons.push('Cancel');
+
+      showActionSheetWithOptions(
+        {
+          cancelButtonIndex: isDeletable ? 2 : 1,
+          destructiveButtonIndex: isDeletable ? 1 : null,
+          options: buttons,
+          title: `${label || abbreviations.address(address, 4, 6)}`,
+        },
+        buttonIndex => {
+          if (buttonIndex === 0) {
+            // Edit wallet
+            renameWallet(wallet_id, address);
+          } else if (isDeletable && buttonIndex === 1) {
+            // Delete wallet
+            deleteWallet(wallet_id, address);
+          }
+        }
+      );
+    },
+    [deleteWallet, renameWallet, selectedWallet, wallets]
+  );
+
+  const onPressAddAccount = useCallback(
+    async wallet_id => {
+      try {
+        await dispatch(createAccountForWallet(wallet_id));
+        await initializeWallet();
+      } catch (e) {
+        logger.log('Error while trying to add account', e);
+      }
+    },
+    [dispatch, initializeWallet]
+  );
+
+  const onPressImportSeedPhrase = useCallback(() => {
+    goBack();
+    navigate(Routes.IMPORT_SEED_PHRASE_SHEET);
+  }, [goBack, navigate]);
+
+  const skeletonTransitionRef = useRef();
+
+  const toggleEditMode = useCallback(() => {
+    setEditMode(!editMode);
+  }, [editMode]);
+
+  useEffect(() => {
+    skeletonTransitionRef &&
```

Comment:
are u sure that this first `skeletonTransitionRef &&` is necessary? im fairly certain that `skeletonTransitionRef.current` always exists